test/cases/regression_tests

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python3
+"""
+Tests the osbuild-image-test by running it against two fake DBs. This test must
+run on Fedora35 x86_64
+"""
+
+import subprocess
+import configparser
+import platform
+import sys
+from subprocess import CalledProcessError
+
+OS_RELEASE_PATH = "/etc/os-release"
+
+config = configparser.ConfigParser()
+with open(OS_RELEASE_PATH, 'r', encoding='utf-8') as f:
+    config_string = '[DEFAULT]\n' + f.read()
+config.read_string(config_string)
+distro = f"{config.get('DEFAULT', 'ID')}-{config.get('DEFAULT', 'VERSION_ID')}"
+distro = distro.replace('"', '')
+distro = distro.replace('.', '')
+
+print(f"Testing the osbuild-image-test for arch {platform.machine()} and "
+        f"distribution {distro} on folder test/data/db-success "
+        "this test must succeed", flush=True)
+
+try:
+    subprocess.run(["tools/osbuild-image-test",
+                    f"--arch={platform.machine()}",
+                    f"--distro={distro}",
+                    "--db-path=test/data/db-success"],
+                    check=True)
+except CalledProcessError:
+    sys.exit(1)
+else:
+    print("SUCCESS")
+
+print(f"Testing the osbuild-image-test for arch {platform.machine()} and "
+        f"distribution {distro} on folder test/data/db-failure "
+        "this test must fail", flush=True)
+
+try:
+    subprocess.run(["tools/osbuild-image-test",
+                    f"--arch={platform.machine()}",
+                    f"--distro={distro}",
+                    "--db-path=test/data/db-failure"],
+                    check=True)
+except CalledProcessError:
+    print("SUCCESS")
+else:
+    sys.exit(1)
```

Comment:
This is probably fine for now but it might be good to have a specific error we can catch to make sure there wasn't some other kind of issue.  Like if the `db-failure` goes away or gets renamed or something else happens that might cause an error in the script, we're be oblivious to it because we're just expecting the script invocation to fail with a non-zero exit code.

Reply:
Yes, I wanted to have proper unit tests at the beginning, but since `osbuild-image-test` is not a python module, you can't really import it to run smarter tests. I could try to setup different return values for the executable and test that I get the right one. Do you have other ideas ?

Reply:
Doesn't need to be too sophisticated I think.  If the test script prints error messages we could just check for the expected one.

Reply:
I'll try to figure out something else better in a follow up then :)